src/handlers/graphical.rs

```diff
@@ -1189,54 +1189,26 @@ impl GraphicalReportHandler {
             .read_span(context_span, self.context_lines, self.context_lines)
             .map_err(|_| fmt::Error)?;
         let context = std::str::from_utf8(context_data.data()).expect("Bad utf8 detected");
-        let mut line = context_data.line();
-        let mut column = context_data.column();
-        let mut offset = context_data.span().offset();
-        let mut line_offset = offset;
-        let mut iter = context.chars().peekable();
-        let mut line_str = String::new();
-        let mut lines = Vec::new();
-        while let Some(char) = iter.next() {
-            offset += char.len_utf8();
-            let mut at_end_of_file = false;
-            match char {
-                '\r' => {
-                    if iter.next_if_eq(&'\n').is_some() {
-                        offset += 1;
-                        line += 1;
-                        column = 0;
-                    } else {
-                        line_str.push(char);
-                        column += 1;
-                    }
-                    at_end_of_file = iter.peek().is_none();
-                }
-                '\n' => {
-                    at_end_of_file = iter.peek().is_none();
-                    line += 1;
-                    column = 0;
-                }
-                _ => {
-                    line_str.push(char);
-                    column += 1;
+        let lines = context
+            .split_inclusive('\n')
+            .enumerate()
+            .map(|(line_number, line)| {
+                let length = line.len();
+                // Strip the newline chars
+                let line = line
+                    .strip_suffix('\n')
+                    .and_then(|line| line.strip_suffix('\r').or(Some(line)))
+                    .unwrap_or(line);
+                Line {
+                    line_number: context_data.line() + line_number + 1,
+                    offset: context_data.span().offset()
+                        + unsafe { line.as_ptr().offset_from(context.as_ptr()) as usize },
```

Comment:
That's the exact opposite of what I wanted to do. The whole reason I went with `offset_from` is to avoid using an `offset` variable, because that variable is a possible source of errors if not computed right (and indeed, such problem in the previous code is why I started to look at this piece of code in the first place).

Also, from a semantic point of view, the "offset from" meaning is what we are looking for ("the offset of line from the beginning of the data buffer"). A `+ line.len()` is only indirectly related to that info. It's only correct if, and only if, the various lines follow each other exactly, without gaps nor overlaps. In particular, it requires the use of `split_inclusive` instead of the more common `split`[^1], creating, IMO, an unnecessary dependency between the split and the offset computation.

And yes it's safe [per design](https://doc.rust-lang.org/stable/reference/types/numeric.html#machine-dependent-integer-types):
>The theoretical upper bound on object and array size is the maximum isize value. This ensures that isize can be used to calculate differences between pointers into an object or array and can address every byte within an object along with one byte past the end.


[^1]: I would have used `split` except that it causes some other issues in some corner cases (I can't remember if it was when the data terminates with a newline or when the data was empty)